code/modules/mob/emote.dm

```diff
@@ -83,3 +90,29 @@
 						riding_datum.force_dismount(M)
 			else
 				L.unbuckle_all_mobs()
+
+/datum/emote/spin/check_cooldown(mob/living/carbon/user, intentional)
+	. = ..()
+	if(.)
+		return
+	if(!can_run_emote(user, intentional=intentional))
+		return
+	if(!iscarbon(user))
+		return
+	var/current_confusion = user.get_confusion()
+	if(current_confusion > BEYBLADE_PUKE_THRESHOLD)
+		user.vomit(BEYBLADE_PUKE_NUTRIENT_LOSS, distance = 0)
```

Comment:
On Discord, arcane reported that the reason why he didn't use projectile vomiting here was that it caused the vomit streams to always be sent North for some reason. That seems like an issue that's worth looking into, either in this PR or in a future one.